test/jobs/validate_dnssec_job_test.rb

```diff
@@ -68,4 +68,77 @@ def test_job_should_not_set_validation_datetime_if_validation_is_invalid
     @domain.reload
     assert_nil @domain.dnskeys.first.validation_datetime
   end
+
+  def test_prepare_validator_configures_dnsruby_resolver_with_correct_parameters
+    job = ValidateDnssecJob.new
+    
+    # To store original environment variable
+    original_timeout = ENV['nameserver_validation_timeout']
+    
+    # Seadista kindel väärtus testiks
+    ENV['nameserver_validation_timeout'] = '4'
+    
+    resolver = job.send(:prepare_validator, "8.8.8.8")
+    
+    assert_instance_of Dnsruby::Resolver, resolver
+    assert resolver.do_validation
+    assert resolver.dnssec
+    assert_equal 4, resolver.packet_timeout
+    assert_equal 4, resolver.query_timeout
+    
+    # Restore original environment variable
+    ENV['nameserver_validation_timeout'] = original_timeout
+  end
+
+  def test_perform_skips_domains_without_nameservers
+    domain = Domain.create!(
+      name: "test.test",
+      registrar: registrars(:bestnames),
+      registrant: @domain.registrant,
+      period: 1,
+      period_unit: 'y',
+      valid_to: 1.year.from_now
+    )
+
+    # Add DNSKEY to domain
+    domain.dnskeys << @dnskey
+
+    # Create a StringIO to capture log output
+    log_output = StringIO.new
+    logger = Logger.new(log_output)
+    logger.level = Logger::INFO
+
+    # Create a job instance and set its logger
+    job = ValidateDnssecJob.new
+    job.define_singleton_method(:logger) { logger }
+
+    # Run the job
+    job.perform(domain_name: domain.name)
+
+    # Verify that the domain was skipped
+    assert_match /No related nameservers for this domain/, log_output.string
```

Comment:
I think you've overcomplicated the test a bit. If you look at the file app/jobs/validate_dnssec_job.rb, the dns key object has a field called validation_datetime, which gets updated when the validation is successful. You could’ve used that for the check.

Of course, if the goal is to verify the logs, then tracking logs makes sense too. But what's really important here is that the job didn’t make any changes to the database record.

We actually need two clear tests:
One test that confirms the table was updated when a nameserver is present.
And this current test should prove that validation was skipped, meaning no changes happened in the table at all.